docs/src/main/tut/typeclasses/comonad.md

```diff
@@ -0,0 +1,87 @@
+---
+layout: docs
+title:  "Comonad"
+section: "typeclasses"
+source: "core/src/main/scala/cats/Comonad.scala"
+scaladoc: "#cats.Comonad"
+---
+# Comonad
+
+Given any type class we can create a "Co" type class for which its functions
+are the "dual" of the other. A dual to a function has the same types but the 
+direction of the arrows are reversed. Whether or not that is useful even possible 
+depends on the particular type.
+
+`Comonad` is a `Functor` and provides duals of the [`Monad`](monad.html) `pure`
+and `flatMap` functions.
+
+### extract
+
+Monad's have `pure` from `Applicative` which gives you the ability to wrap 
+a value `A` using the type constructor giving an `F[A]`. Comonad has 
+`extract` which instead takes an `F[A]` and extracts the `A`. Therefore, to be 
+able to implement extract we must have a type of which we are certain
+we can get an `A` from an `F[A]`. For example we cannot always get an `A` 
+from a `List[A]` because if the list is empty there is nothing to get.
+
+For the same reason, `Option` doesn't have a Comand instance, because we 
+cannot always get an `A` from an Option, it may be empty too.
+
+Some examples that we can implement `Comonad` for include `OneAnd`, `Tuple2` 
+and the "non empty" collections.
+
+First some imports.
+
+```tut:silent
+import cats._
+import cats.data._
+import cats.implicits._
+import cats.syntax.comonad._
+import cats.instances.list._
+```
+
+`NonEmptyList` has a `Comonad` instance and its implementation of `extract` 
+simply returns the head element of the list, which we know we will always
+have.
+
+```tut:book
+NonEmptyList.of(1,2,3).extract
+```
+
+### coflatMap
+
+`coflatMap` is the dual of Monad's `flatMap`.  While `flatMap` allows us to chain 
+together operations in a monadic context, `CoflatMap` takes a value in some context
```

Comment:
`CoflatMap` -> `coflatMap` ?

Reply:
I tried to match the structure of the monad page... these extract and coflatMap sections are lowercase to match the function names, to distinguish them from the CoflatMap type heading later. I can change it if it’s confusing

Reply:
oh yeah I see what you mean will fix